cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/RunRecordTimeToLiveService.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.services;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+import com.google.inject.Inject;
+import io.cdap.cdap.common.conf.CConfiguration;
+import io.cdap.cdap.common.conf.Constants;
+import io.cdap.cdap.internal.app.store.AppMetadataStore;
+import io.cdap.cdap.spi.data.transaction.TransactionException;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service which periodically scans the database tables for run records which should be deleted per
+ * the global time to live value.
+ *
+ * <p>Does not run if no TTL is configured or a TTL of 0 is specified.
+ */
+public final class RunRecordTimeToLiveService extends AbstractIdleService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(RunRecordTimeToLiveService.class);
+
+  private final TransactionRunner transactionRunner;
+  private final boolean enabled;
+  private final Duration ttlMaxAge;
+  private final Duration checkFrequency;
+  private final Duration initialDelay;
+  private final Clock clock;
+
+  private ScheduledExecutorService service;
+
+  @Inject
+  public RunRecordTimeToLiveService(CConfiguration cConf, TransactionRunner transactionRunner) {
```

Comment:
normally we keep injected constructors package private, as we don't want anything to manually create it except maybe in unit tests.

Reply:
Done